hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/Invocation.java

```diff
@@ -487,11 +493,15 @@ private void handleRetry(Object cause) {
         } else {
             context.invocationRegistry.deregister(this);
 
-            if (invokeCount < MAX_FAST_INVOCATION_COUNT) {
-                // fast retry for the first few invocations
-                context.asyncExecutor.execute(new InvocationRetryTask());
-            } else {
-                context.executionService.schedule(ASYNC_EXECUTOR, new InvocationRetryTask(), tryPauseMillis, MILLISECONDS);
+            try {
+                if (invokeCount < MAX_FAST_INVOCATION_COUNT) {
+                    // fast retry for the first few invocations
+                    context.asyncExecutor.execute(new InvocationRetryTask());
+                } else {
+                    context.executionService.schedule(ASYNC_EXECUTOR, new InvocationRetryTask(), tryPauseMillis, MILLISECONDS);
+                }
+            } catch (RejectedExecutionException e) {
+                complete(e);
```

Comment:
I see. wouldn't be  `HazelcastNotActiveException` more suitable*? It says what's going on, what's the cause. 

*assuming it can happen only during shutdown.